hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/AbstractEntityResultGraphNode.java

```diff
@@ -131,33 +120,42 @@ public AbstractEntityResultGraphNode(
 		}
 	}
 
-	private void visitCompositeIdentifierMapping(
-			NavigablePath navigablePath,
+	private void visitIdentifierMapping(
+			EntityIdentifierNavigablePath navigablePath,
 			DomainResultCreationState creationState,
 			EntityIdentifierMapping identifierMapping,
 			TableGroup entityTableGroup) {
-		ManagedMappingType mappingType = (ManagedMappingType) identifierMapping.getPartMappingType();
-		fetches = new ArrayList<>();
-		mappingType.visitAttributeMappings(
-				attributeMapping -> {
-					if ( attributeMapping instanceof ToOneAttributeMapping ) {
-						( (ToOneAttributeMapping) attributeMapping ).getForeignKeyDescriptor().createDomainResult(
-								new EntityIdentifierNavigablePath( navigablePath ),
-								entityTableGroup,
-								null,
-								creationState
-						);
-					}
-					else {
-						attributeMapping.createDomainResult(
-								new EntityIdentifierNavigablePath( navigablePath ),
-								entityTableGroup,
-								null,
-								creationState
-						);
+		final MappingType mappingType = identifierMapping.getPartMappingType();
+		if ( mappingType instanceof ManagedMappingType ) {
+			( (ManagedMappingType) mappingType ).visitAttributeMappings(
+					attributeMapping -> {
+						if ( attributeMapping instanceof ToOneAttributeMapping ) {
+							( (ToOneAttributeMapping) attributeMapping ).getForeignKeyDescriptor().createDomainResult(
+									navigablePath,
+									entityTableGroup,
+									null,
+									creationState
+							);
+						}
+						else {
+							attributeMapping.createDomainResult(
+									navigablePath,
+									entityTableGroup,
+									null,
+									creationState
+							);
+						}
```

Comment:
It might be minor, but the above code involves some duplication and might be simplified as
```
attributeMapping -> {
    final AttributeMapping realMapping;
    if ( attributeMapping instanceof ToOneAttributeMapping ) {
         realMapping = ( (ToOneAttributeMapping) attributeMapping ).getForeignKeyDescriptor();
    }
    else {
         realMapping = attributeMapping;
    }
    realMapping.createDomainResult(
        navigablePath,
        entityTableGroup,
        null,
        creationState
    );
}
```